kamu-adapter-http/src/http_server_middleware.rs

```diff
@@ -73,6 +73,12 @@ pub struct DatasetResolverMiddleware<Svc, IdExt, Extractor> {
     layer: DatasetResolverLayer<IdExt, Extractor>,
 }
 
+impl<Svc, IdExt, Extractor> DatasetResolverMiddleware<Svc, IdExt, Extractor> {
+    fn is_dataset_optional(request: &Request<Body>) -> bool {
+        "/push" == request.uri().path()
```

Comment:
Will this fail to match under a different mount point, e.g. `/:dataset_id`? Or is axum giving us request URI relative to the mount?

Reply:
As you can see, we don't have dataset id here as well. We'll test of course when it's time.